exporters/metadata.go

```diff
@@ -0,0 +1,12 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package exporters
+
+// Metadata represents a set of exporter specific properties
+type Metadata struct {
+	Properties map[string]interface{} `json:"properties"`
+	Buffer     *string                `json:"-"`

```

Comment:
Why is this a pointer of a string?

Reply:
This is to allow passing in a string buffer that is under control of the caller. This is necessary to make the component testable - see unit tests.